### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by NikitaMihailyuk**
all files diplom 
and new:
work after recommendation:

Add basebrowserpage
add login assert
move model folder
asserthelper was deleted
small improves of code

### CODE REVIEW

File: BussinessObject/ApiObjects/Services/TestCaseService.cs [hunk 1]
```diff
@@ -4,82 +4,98 @@
...
 
+        //https://support.testrail.com/hc/en-us/articles/15758177606676-API-uses-cases-intro
+        /*     
+Case fields	GET index.php?/api/v2/get_case_fields
+Case templates	GET index.php?/api/v2/get_templates/{project_id}
+Case priorities	GET index.php?/api/v2/get_priorities
+Case types	GET index.php?/api/v2/get_case_types
+Case statuses	GET index.php?/api/v2/get_case_statuses
+  "https://example.testrail.io/index.php?/api/v2/add_case/432"
+         */
```
Comment (VladimirVolkovVD) [inline BussinessObject/ApiObjects/Services/TestCaseService.cs]: delete

File: BussinessObject/ApiObjects/Services/TestCaseService.cs [hunk 2]
```diff
@@ -4,82 +4,98 @@
...
+        TestCaseBuilder testCaseBuilder = new TestCaseBuilder();
 
-        public RestResponse GetRuns(string caseCode)
+        [AllureStep]
+        public RestResponse GetRuns(int offset)
         {
-            var request = new RestRequest(GetCaseByCodeEndpoint, Method.Get).AddUrlSegment("code", caseCode); ;
-            request.AddBody(caseCode);
+            var request = new RestRequest(GetRunsByCodeEndpoint, Method.Get).AddUrlSegment("code", offset); ;
+            request.AddBody(offset);
```
Comment (VladimirVolkovVD) [inline BussinessObject/ApiObjects/Services/TestCaseService.cs]: offset or code?

File: BussinessObject/BrowserObjects/PageObjects/BaseBrowserPage.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            Assert.AreEqual(errorTopString, errorTop);
+            Assert.AreEqual(errorString, errorText);
+        }
+
+        public void AssertUserName(string userName)
+        {
+            string cuurentUserName=userNameElement.GetElement().Text;
+            Assert.AreEqual(cuurentUserName, userName);
+        }
+        //*[@class='navigation-username']
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/BaseBrowserPage.cs]: ?

File: BussinessObject/BrowserObjects/PageObjects/MainPage.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        return new MainPage();
+    }
+    [AllureStep]
+    public TestCasePage OpenTestCasePage()
+    {
+        projectForUpdate.GetElement().Click();
+        testCaseButton.GetElement().Click(); 
+        return new TestCasePage();
+    }
+}
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/MainPage.cs]: empty line after method fix code style

File: BussinessObject/BrowserObjects/PageObjects/TestCasePage.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            custom_preconds_display.GetElement().SendKeys(testCaseData.Custom_preconds);
+            custom_steps_display.GetElement().SendKeys(testCaseData.Custom_steps_separated.ToString());
+            custom_expected_display.GetElement().SendKeys(testCaseData.Custom_expected);
+            saveButton.GetElement().Click();    
+        }
+        [AllureStep]
+        public void DeleteTestCase()
+        {
+     
+        action.MoveToElement(testCaseElement.GetElement()).Perform();
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/TestCasePage.cs]: fix code style

**Review by VladimirVolkovD — COMMENTED** [review]
Minor changes need

**Comment by NikitaMihailyuk** [comment]
Added  

some code style fixed
config account data

File: BussinessObject/ApiObjects/Services/ProjectService.cs [hunk 6]
```diff
@@ -1,71 +1,67 @@
...
         {
-            var request = new RestRequest(ProjectAcces, Method.Delete);
-            request = request.AddUrlSegment("code", code).AddParameter("application/json", "{\"member_id\":" + memberid + "}", ParameterType.RequestBody);
-            Console.WriteLine(request);
+            var request = new RestRequest(AddProjectEndpoint, Method.Post);
+            var body = projectBuilder.CreateProject();
+            request.AddBody(body);
+
             return apiClient.Execute(request);
         }
```
Comment (VladimirVolkovVD) [inline BussinessObject/ApiObjects/Services/ProjectService.cs:58]: empty line need

File: BussinessObject/BrowserObjects/PageObjects/BaseBrowserPage.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        public void LoginErrorAssert()
+        {
+            string errorTopString = DataHelper.commonErrorMessage;
+            string errorString = DataHelper.errorInputLoginMessage;
+            string errorTop = messageErrorTopString.GetElement().Text;
+            string errorText = messageErrorText.GetElement().Text;
+            Assert.AreEqual(errorTopString, errorTop);
+            Assert.AreEqual(errorString, errorText);
+        }
+
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/BaseBrowserPage.cs:32]: extra empty line

File: BussinessObject/BrowserObjects/PageObjects/BaseBrowserPage.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+        public void AssertUserName()
+        {
+            var userName = Configuration.Account.ToString();
+            string cuurentUserName = userNameElement.GetElement().Text;
+            Assert.AreEqual(cuurentUserName, userName);
+        }
+
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/BaseBrowserPage.cs:39]: extra empty line

File: BussinessObject/BrowserObjects/PageObjects/LoginPage.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+using Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Allure.Attributes;
+using NLog;
+using BussinessObject.BrowserObjects.PageObjects;
+
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/LoginPage.cs:12]: extra empty line

File: BussinessObject/BrowserObjects/PageObjects/MainPage.cs [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    [AllureStep]
+    public ProjectPage OpenEditProjectPage()
+    {
+        projectForUpdate.GetElement().Click();
+        editeProjectButton.GetElement().Click();
+
+        return new ProjectPage();
+    }
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/MainPage.cs:55]: need empty line

File: BussinessObject/BrowserObjects/PageObjects/MainPage.cs [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        Browser.Instance.NavigateToUrl(DataHelper.projectPageUrl);
+        logger.Info($"Navigate to url {DataHelper.projectPageUrl}");
+
+        deleteProjectIcon.GetElement().Click();
+        dialogDeleteButton.GetElement().Click();
+        dialogAcceptDeleteButton.GetElement().Click();
+
+        return new MainPage();
+    }
+    [AllureStep]
```
Comment (VladimirVolkovVD) [inline BussinessObject/BrowserObjects/PageObjects/MainPage.cs:70]: extra empty line

File: BussinessObject/UserBuilder.cs [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+        public static UserModel GetStandartUserWithoutName()
+        {
+            return new UserModel
+            {
+                Name = " ",
+                Password = TestContext.Parameters.Get("StandartUserPassword"),
+            };
+        }
+
```
Comment (VladimirVolkovVD) [inline BussinessObject/UserBuilder.cs:33]: extra empty line

File: BussinessObject/UserBuilder.cs [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+        public static UserModel GetTestRailUser()
+        {
+            return new UserModel
+            {
+             Name = Configuration.Browser.ListOfUsers[0].Name,
+             Password = Configuration.Browser.ListOfUsers[0].Password
+            };
+        }
+
```
Comment (VladimirVolkovVD) [inline BussinessObject/UserBuilder.cs:24]: extra empty line

File: BussinessObject/UserBuilder.cs [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+        public static UserModel GetRandomUser()
+        {
+            return new UserModel
+            {
+                Name = Faker.Internet.Email(),
+                Password = Faker.Internet.Password(),
+            };
+        }
+
```
Comment (VladimirVolkovVD) [inline BussinessObject/UserBuilder.cs:42]: extra empty line

File: Core/BaseApiClient.cs [hunk 15]
```diff
@@ -23,7 +23,7 @@ public BaseApiClient(string url)
             {
```
Comment (VladimirVolkovVD) [inline Core/BaseApiClient.cs:23]: MaxTimeout = 10000,

 duplicate moove to the config time = 10000

**Review by VladimirVolkovD — COMMENTED** [review]
let's do it

**Pull request merged**